tests/main/preseed-lxd/task.yaml

```diff
@@ -0,0 +1,102 @@
+summary: Check that preseeding of current ubuntu cloud image works under lxd.
+description: |
+  This test checks that preseeding of Ubuntu cloud images with snap-preseed
+  command works in lxc container. 
+
+# this test works only on 18.04 because it requires lxd from deb (lxd snap
+# as it wouldn't allow mount) and tries to replicate launchpad builder setup.
+systems: [ubuntu-18.04-*]
+
+environment:
+  IMAGE_MOUNTPOINT: /mnt/cloudimg
+
+prepare: |
+  IMGURL="https://cloud-images.ubuntu.com/focal/current/focal-server-cloudimg-amd64.img"
+  wget --quiet "$IMGURL" -O cloudimg.img
+
+  apt install -y qemu-utils
+  modprobe nbd
+
+  # Run qemu-ndb as a service, so that it does not interact with ssh
+  # stdin/stdout it would otherwise inherit from the spread session.
+  systemd-run --system --service-type=forking --unit=qemu-ndb-preseed.service "$(command -v qemu-nbd)" --fork -c /dev/nbd0 "$(pwd)/cloudimg.img"
+  # nbd0p1 may take a short while to become available
+  retry-tool -n 5 --wait 1 test -e /dev/nbd0p1
+  mkdir -p "$IMAGE_MOUNTPOINT"
+  mount /dev/nbd0p1 "$IMAGE_MOUNTPOINT"
+
+  # Inject current snapd into seeds
+  #shellcheck source=tests/lib/snaps.sh
+  . "$TESTSLIB"/snaps.sh
+  SNAP_IMAGE=$(ls "$IMAGE_MOUNTPOINT"/var/lib/snapd/seed/snaps/snapd_*.snap)
+  unsquashfs "$SNAP_IMAGE"
+  cp /usr/lib/snapd/snapd squashfs-root/usr/lib/snapd/snapd
+  rm "$SNAP_IMAGE"
+  mksnap_fast squashfs-root "$SNAP_IMAGE"
+  sed -i "$IMAGE_MOUNTPOINT/var/lib/snapd/seed/seed.yaml" -E -e "s/^(\\s+)name: snapd/\\1name: snapd\\n\\1unasserted: true/"
+
+restore: |
+  lxc stop my-ubuntu --force
+  lxc delete my-ubuntu
+
+  umount "$IMAGE_MOUNTPOINT"
+  rmdir "$IMAGE_MOUNTPOINT"
+
+  # qemu-nbd -d may sporadically fail when removing the device,
+  # reporting it's still in use.
+  retry-tool -n 5 --wait 1 qemu-nbd -d /dev/nbd0
+
+  lxd-tool undo-lxd-mount-changes
+
+execute: |
+  echo "Install snapd"
+  # shellcheck disable=SC2046
+  SNAPD_DEB=$(basename $(ls "$GOHOME"/snapd_*.deb))
+  dpkg -i "$GOHOME/$SNAPD_DEB"
+
+  echo "Install lxd"
+  apt install -y lxd
```

Comment:
same about using pkgdb helper for installing. Also it is needed to remove it in case the package is installed in the test.

Reply:
In this case I think this needs to be here. I don't want lxd installed by default. Also, I'm deliberately forcing lxd deb in this test, while we install lxd snap in other tests. Installing  deb by default could cause problems in other tests. I've added apt purge to the restore section.

Reply:
Makes sense, so I think best way to do it is to do:
if is installed lxd then touch remove-lxd && apt install lxd; if
Then you remove lxd just if fle remove-lxd exists